File: src/main/java/calebabg/abstractions/UriAudioSource.java
package calebabg.abstractions;

public class UriAudioSource implements IAudioResource {
    private final String id;
    private final String title;
    private final String resourcePath;

    public UriAudioSource(String id, String title, String resourcePath) {
        this.id = id;
        this.title = title;
        this.resourcePath = resourcePath;
    }

    public String id() {
        return id;
    }

    public String title() {
        return title;
    }

    public String mediaPath() {
        return resourcePath;
    }

    public AudioResourceType resourceType() {
        return AudioResourceType.URI_RESOURCE;
    }
}

File: src/main/kotlin/com/calebabg/abstractions/UriAudioSource.kt
package com.calebabg.abstractions

class UriAudioSource(private val id: String, private val title: String, private val resourcePath: String) : AudioResource {
    override fun id(): String = id
    override fun title(): String = title
    override fun mediaPath(): String = resourcePath
    override fun resourceType(): AudioResourceType = AudioResourceType.URI_RESOURCE
}